client/src/test/scala/dwolla/cloudflare/AccessControlRuleClientSpec.scala

```diff
@@ -0,0 +1,338 @@
+package dwolla.cloudflare
+
+import cats.effect.Sync
+import com.dwolla.cloudflare._
+import com.dwolla.cloudflare.domain.model.Exceptions.UnexpectedCloudflareErrorException
+import com.dwolla.cloudflare.domain.model.{AccountId, AccountIdTag}
+import com.dwolla.cloudflare.domain.model.accesscontrolrules._
+import io.circe.Json
+import org.http4s.Status
+import org.http4s.client.Client
+import org.specs2.mutable.Specification
+import org.specs2.specification.Scope
+import io.circe.literal._
+
+class AccessControlRuleClientSpec extends Specification {
+
+  trait Setup extends Scope {
+    val accountId: AccountId = shapeless.tag[AccountIdTag][String]("account-id")
+
+    val authorization = CloudflareAuthorization("email", "key")
+    val fakeService = new FakeCloudflareService(authorization)
+  }
+
+  "list" should {
+    "list challenge rules" in new Setup {
+      val http4sClient = fakeService.client(fakeService.listAccessRules(Map(1 → SampleResponses.Successes.listAccessRulesPage1, 2 → SampleResponses.Successes.listAccessRulesPage2, 3 → SampleResponses.Successes.listAccessRulesPage3), accountId))
+      val client = buildAccessControlRuleClient(http4sClient, authorization)
+
+      val output: List[Rule] = client
+        .list(accountId)
+        .compile
+        .toList
+        .unsafeRunSync()
+      output must be_==(
+        List(
```

Comment:
Does the order of the response matter here? If not, it'd be less brittle to change this `output must be_==(List(…` to one `output must contain` for each rule in the output, e.g.:

```scala
output must contain(Rule(
  id = "fake-access-rule-1",
  notes = Some("Some notes"),
  allowedModes = List("whitelist", "block", "challenge", "js_challenge"),
  mode = Some("challenge"),
  configuration = RuleConfiguration("ip", "1.2.3.4"),
  createdOn = Some("2014-01-01T05:20:00.12345Z"),
  modifiedOn = Some("2014-01-01T05:20:00.12345Z"),
  scope = RuleScope("fake-rule-scope", "Some Account", "account"),
))

output must contain({rule2})
…
```